### ISSUE TITLE
[Feature]: Separate `launchPersistentContext` argument to set initial URL

### ISSUE THREAD

**Issue by dunielpls** [microsoft/playwright#31311]
### 🚀 Feature Request

In some cases it is necessary to launch the browser with an initial URL in the command-line arguments.

How it currently seems to work is that `--no-startup-window` is removed and `about:blank` appended to the command-line arguments when `launchPersistentContext` is used:
https://github.com/microsoft/playwright/blob/a9200be0aff9efb4eea7bf0a786760fe10e74f15/packages/playwright-core/src/server/chromium/chromium.ts#L287-L290

This is a bit problematic for me as `about:blank` first loads and is visible, then I have to grab the first page in the new persistent context and navigate that to the desired URL.

Implementing the feature should however be fairly simple and I may submit a pull request soon.

### Example

My use case is digital signage. The `--no-startup-window` set when `launch` is used does not work properly with `--kiosk` due to Chromium finding it a "non-sensical combination of options" [sic], and it seems like it is impossible to create a new page in a new context and make it behave like normal kiosk windows.

### Motivation

I find it way easier to launch the browser directly with a user-supplied URL that is navigated to in the initial page rather than rendering `about:blank`, finding the first page, and navigating it to a new URL. I can imagine that other people with "slightly odd" use cases would find this ergonomic.

**Comment by dunielpls** [comment]
It may also be desirable to allow setting an initial URL for non-persistent contexts too, i.e. making it possible to have the default context automatically navigate to a user-supplied URL when using `launch`.

This would require removing the `--no-startup-window` flag when `isPersistent` is false and appending the URL in question too.

**Comment by yury-s** [comment]
This parameters you pass in launch/launchPersistentContext are fully customizable with a combination of [ignoreDefaultArgs](https://playwright.dev/docs/api/class-browsertype#browser-type-launch-persistent-context-option-ignore-default-args) and [args](https://playwright.dev/docs/api/class-browsertype#browser-type-launch-persistent-context-option-args). We settled over time on the default arguments that we have today, they give you the most flexible control over the browser from the very beginning of the process run, there are no plans to change it.

**Comment by dunielpls** [comment]
That is nearly true - it isn't possible to 'specify page to be opened', which seems like a fairly arbitrary limitation.

https://github.com/microsoft/playwright/blob/a9200be0aff9efb4eea7bf0a786760fe10e74f15/packages/playwright-core/src/server/chromium/chromium.ts#L301-L302

**Comment by dunielpls** [comment]
For anyone else looking into something similar, I grepped through parts of the Chromium source and it appears that the kiosk restrictions remain present even with both of the flags `--kiosk` and `--no-startup-window` - the only problem is that Chromium has no idea how to make a non-existent window full-screen.

To solve that I ended up opening a CDP session and getting the `windowId` with `Browser.getWindowForTarget` before requesting fullscreen with `Browser.setWindowBounds` (parameters `{windowId, bounds: {windowState: "fullscreen"}}`.

Despite solving my issue through a sort of workaround I still wonder why Playwright by default disallows arguments that do not start with a hyphen, i.e. opening a specific URL in the default context on startup.
